tests/test_more.py

```diff
@@ -4537,3 +4537,55 @@ def test_basic(self):
         self.assertEqual(it.items_seen, 1)
         self.assertEqual(''.join(it), '123456789')
         self.assertEqual(it.items_seen, 10)
+
+
+class ChunkedEvenTests(TestCase):
+    """Tests for ``chunked_even()``"""
+
+    def test_0(self):
+        self._test_finite('', 3, [])
+
+    def test_1(self):
+        self._test_finite('A', 1, [['A']])
+
+    def test_4(self):
+        self._test_finite('ABCD', 3, [['A', 'B'], ['C', 'D']])
+
+    def test_5(self):
+        self._test_finite('ABCDE', 3, [['A', 'B', 'C'], ['D', 'E']])
+
+    def test_6(self):
+        self._test_finite('ABCDEF', 3, [['A', 'B', 'C'], ['D', 'E', 'F']])
+
+    def test_7(self):
+        self._test_finite('ABCDEFG', 3, [['A', 'B', 'C'], ['D', 'E'], ['F', 'G']])
+
+    def _test_finite(self, seq, n, expected):
+        # Check with and without `len()`
+        self.assertEqual(list(mi.chunked_even(seq, n)), expected)
+        self.assertEqual(list(mi.chunked_even(iter(seq), n)), expected)
+
+    def test_infinite(self):
+        n = 4
+        def limited_count():
```

Comment:
Can this be replaced with `islice(count(), 0, n * k + n * n)`?

Reply:
`k` varies. This test is kind of like a "timeout", it's supposed to check that, if given an infinite iterable, it can (1) yield things without reading the whole sequence (look-ahead is finite), (2) yield things without reading too far ahead (look-ahead < k^2).